### ISSUE TITLE
Issue with arrow functions, destructuring, and argument scope - arguments from parent scope used

### ISSUE THREAD

**Issue by bluepnume** [codemix/flow-runtime#47]
This is a:

 - [x] Bug Report
 - [ ] Feature Request
 - [ ] Question
 - [ ] Other

Which concerns:

 - [x] flow-runtime
 - [x] babel-plugin-flow-runtime
 - [ ] flow-runtime-validatosr
 - [ ] flow-runtime-mobx
 - [ ] flow-config-parser
 - [ ] The documentation website

---

### What is the current behaviour?

I have the following code:

```
type Person = {
  name: string
};

let sayHello = ({ name } : Person) => {
  sayHello(name);
}

sayHello({ name: 'Kermit' });
```

This compiles to:

```
var _arguments = arguments;
import t from 'flow-runtime';
const Person = t.type('Person', t.object(t.property('name', t.string())));

let sayHello = t.annotate(function ({ name }) {
  t.param('arguments[0]', Person).assert(_arguments[0]);

  sayHello(name);
}, t.function(t.param('_arg', Person)));

sayHello({ name: 'Kermit' });
```

And gives the error:

```
 arguments[0].name must be a string

Expected: string

Actual: void
```

The problem is, since arguments is not defined in the arrow function, it's checking against `_arguments`, which is `arguments` from the parent scope. And obviously that doesn't work because the parent scope's arguments don't match what is expected in `sayHello`

---

### What is the expected behaviour?

There should be no type error, and the function should correctly check the arguments for the current scope.

---

### Which package versions are you using?

- babel-plugin-flow-runtime@0.3.0
- flow-runtime@0.3.0

**Comment by bluepnume** [comment]
Thanks for the fix! Love the tool, keep up the awesome work.

**Comment by phpnode** [comment]
your welcome :) fix is in 0.4.0
